packages/nx/src/utils/ignore.spec.ts

```diff
@@ -0,0 +1,89 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { createTree } from '../generators/testing-utils/create-tree';
+import { Tree } from '../generators/tree';
+import {
+  createCombinedIgnore,
+  createIgnoreFromFS,
+  createIgnoreFromTree,
+} from './ignore';
+import { stripIndents } from './strip-indents';
+
+describe('createIgnoreFromFS', () => {
+  let tmpDir: string;
+
+  beforeEach(() => {
+    tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'nx-createIgnoreFromFS-'));
+  });
+
+  afterEach(() => {
+    fs.rmSync(tmpDir, { recursive: true });
+  });
```

Comment:
Can we use memfs for this instead? It would be best to avoid actually writing to disk in here. Check `packages/nx/src/config/workspaces.spec.ts` for an example